checks/hsts.go

```diff
@@ -0,0 +1,98 @@
+package checks
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+const (
+	StrictTransportSecurity = "Strict-Transport-Security"
+	includeSubDomains       = "includeSubDomains"
+	preload                 = "preload"
+	NilHeadersError         = "Site does not serve any HSTS headers."
+	MaxAgeError             = "HSTS max-age is less than 10886400."
+	SubdomainsError         = "HSTS header does not include all subdomains."
+	PreloadError            = "HSTS header does not contain the preload directive."
+	HstsSuccess             = "Site is compatible with the HSTS preload list!"
+)
+
+type HSTSResponse struct {
+	Message    string `json:"message"`
+	Compatible bool   `json:"compatible"`
+	HSTSHeader string `json:"hstsHeader"`
+}
+
+type Hsts struct {
+	client *http.Client
+}
+
+func NewHsts(client *http.Client) *Hsts {
+	return &Hsts{client: client}
+}
+
+func (h *Hsts) Validate(ctx context.Context, url string) (*HSTSResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := h.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	hstsHeader := resp.Header.Get(StrictTransportSecurity)
```

Comment:
headers can be set in one valid string such as 
```
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
```
or multiple such as
```
Strict-Transport-Security: max-age=63072000
Strict-Transport-Security: includeSubDomains; preload
```

both are valid but the use of `Get` only returns one in the case of scenario 2.

Reply:
*http.Response.header is a map so would the second `Strict-Transport-Security` not replace the first value?